docs/site/tutorials/authentication/Authentication-Tutorial.md

```diff
@@ -0,0 +1,912 @@
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/authentication-tutorial.html
+summary: A LoopBack 4 application that uses authentication
+---
+
+## Overview
+
+LoopBack 4 has an authentication package `@loopback/authentication` which allows
+you to secure your API endpoints with custom authentication strategies that
+implement a common interface and are registered via an extensionPoint/extension
+pattern.
+
+## Try it out
+
+If you'd like to see the final results of this tutorial as an example
+application, follow these steps:
+
+1.  Clone the `strongloop/loopback4-example-shopping` repository:
+
+    ```sh
+    git clone git@github.com:strongloop/loopback4-example-shopping.git
+    ```
+
+2.  Switch to the directory.
+
+    ```sh
+    cd loopback4-example-shopping
+    ```
+
+3.  Install all the dependencies.
+
+    ```sh
+    npm install
+    ```
+
+4.  Start the MongoDB and the Redis containers:
+
+    ```sh
+    npm run docker:start
+    ```
+
+5.  Build and start the application
+
+    ```sh
+    npm start
+
+    Recommendation server is running at http://127.0.0.1:3001.
+    Server is running at http://[::1]:3000
+    Try http://[::1]:3000/ping
+    ```
+
+6.  In a browser, navigate to [http://[::1]:3000](http://127.0.0.1:3000) or
+    [http://127.0.0.1:3000](http://127.0.0.1:3000), and click on `/exporer` to
+    open the `API Explorer`.
+
+7.  In the `UserController` section, click on `POST /users`, click on
+    `'Try it out'`, specify:
+
+    ```ts
+    {
+    "id": "1",
+    "email": "user1@example.com",
+    "password": "thel0ngp@55w0rd",
+    "firstName": "User",
+    "lastName": "One"
+    }
+    ```
+
+    and click on `'Execute'` to **add** a new user named `'User One'`.
+
+8.  In the `UserController` section, click on `POST /users/login`, click on
+    `'Try it out'`, specify:
+
+    ```ts
+    {
+      "email":    "user1@example.com",
+      "password": "thel0ngp@55w0rd"
+    }
+    ```
+
+    and click on `'Execute'` to **log in** as `'User One'`.
+
+    A JWT token is sent back in the response.
+
+    For example:
+
+    ```ts
+    {
+      "token": "some.token.value"
+    }
+    ```
+
+9.  Perform a `GET` on the secured endpoint `/users/me` making sure to provide
+    the JWT token in the `Authorization` header. If authentication succeeds, the
+    [user profile](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/types.ts)
+    of the currently authenticated user will be returned in the response. If
+    authentication fails due to a missing/invalid/expired token, an
+    [HTTP 401 UnAuthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401)
+    is thrown.
+
+    ```sh
+    curl -X GET \
+    --header 'Authorization: Bearer some.token.value' \
+    http://127.0.0.1:3000/users/me
+    ```
+
+    The response is :
+
+    ```sh
+    {"id":"1","name":"User One"}
+    ```
+
+## Adding Authentication to a LoopBack 4 Application
```

Comment:
1. Can you describe how authentication is needed for the shopping example?
2. Can you get one of the architectural diagrams to help our users establish the big picture first?

Reply:
@raymondfeng 

> Can you describe how authentication is needed for the shopping example?
> Can you get one of the architectural diagrams to help our users establish the big picture first?

I think my new intro paragraph and JWT diagram gives a decent idea.  At the end of the document I point them to the `Authentication Component` document. It has architectual diagrams.  I think it is best not to duplicate in many places. 

Let me know if my new intro and diagram are sufficient. thanks.